cmd/virtual-kubelet/root/root.go

```diff
@@ -102,6 +102,28 @@ func runRootCommand(ctx context.Context, c *Opts) error {
 		return err
 	}
 
+	eb := record.NewBroadcaster()
+	eb.StartRecordingToSink(&corev1clients.EventSinkImpl{Interface: localClient.CoreV1().Events(corev1.NamespaceAll)})
+
+	// active-passive leader election; blocking leader election.
+	// here we have multiple virtual kubelet pods running for the same virtual node.
+	// We want to avoid that multiple virtual kubelet pods reflect the same resources.
+	if leader, err := leaderelection.Blocking(ctx, localConfig, eb, &leaderelection.Opts{
+		PodInfo: leaderelection.PodInfo{
+			PodName:   c.PodName,
+			Namespace: c.TenantNamespace,
+		},
+		LeaderElectorName: c.NodeName,
+		LeaseDuration:     c.VirtualKubeletLeaseLeaseDuration,
+		RenewDeadline:     c.VirtualKubeletLeaseRenewDeadline,
+		RetryPeriod:       c.VirtualKubeletLeaseRetryPeriod,
+	}); err != nil {
+		return err
+	} else if !leader {
+		klog.Info("This virtual-kubelet is not the leader")
```

Comment:
Isn't here more correct logging an error? If we exit the blocking function and the virtual kubelet is not the leader maybe something went wrong, isn't it?